modules/components/src/Aggs/TermAgg.js

```diff
@@ -120,7 +121,11 @@ class TermAggs extends React.Component {
                             '-',
                           )}`}
                         />
-                        {bucket.name}
+                        {!valueCharacterLimit
```

Comment:
You can use lodash truncate to avoid these checks. You could even do something like 
```
truncate(str, {length: valueCharacterLimit  || Infinity})
```
to default to the entire value :)